test/cpp/interop/interop_client.h

```diff
@@ -35,17 +38,25 @@ typedef std::function<void(const InteropClientContextInspector&,
                            const SimpleRequest*, const SimpleResponse*)>
     CheckerFn;
 
-typedef std::function<std::shared_ptr<Channel>(void)> ChannelCreationFunc;
+typedef std::function<std::shared_ptr<Channel>()> ChannelCreationFunc;
+
+typedef std::function<std::shared_ptr<Channel>(
```

Comment:
Instead of using a function that takes another function as a parameter, I think it would be simpler to just pass in the channel args from the caller, like this:

```
typedef std::function<std::shared_ptr<Channel>(ChannelArguments)>
    ChannelCreationFuncWithCustomArgs;
```

Then the new `InteropClient` ctor can look like this:

```
InteropClient::InteropClient(
    ChannelCreationFuncWithCustomArgs channel_creation_func,
    bool new_stub_every_test_case, bool do_not_abort_on_transient_failures)
    : serviceStub_(
          [&]() {
            InitializeCustomLbPolicyIfNeeded();
            ChannelArguments args;
            load_report_tracker_->SetupOnChannel(&args);
            return channel_creation_func(std::move(args));
          },
          new_stub_every_test_case),
      do_not_abort_on_transient_failures_(do_not_abort_on_transient_failures) {}
```

And the lambdas in client.cc can just add their own args to the passed-in args.

Reply:
I don't understand what you mean about not wanting to bury the service config flag.  The approach I proposed here would not require changing where the flag is handled; that would still happen in the lambdas that are created in client.cc.